scripts/yarn-lock-scan.js

```diff
@@ -0,0 +1,16 @@
+const fs = require('fs');
+const path = require('path');
+
+const yarnLockPath = './yarn.lock';
+
+try {
+  const data = fs.readFileSync(path.resolve(yarnLockPath), 'utf8');
+  if (data.match(/artifactory/g)) {
+    console.error(
+      "Error: Artifactory references in your yarn.lock! Please make sure you're using the official yarn or npm registry when downloading your dependencies!",
+    );
+    process.exit(1);
+  }
+} catch (e) {
+  console.error('Error', e.stack);
```

Comment:
if smth goes wrong during `readFileSync`, we need to fail the script, but this will prevent it from failing and only output the erro